src/auth/github-auth.service.ts

```diff
@@ -0,0 +1,121 @@
+import {
+    Injectable,
+    Inject,
+    InternalServerErrorException,
+    NotFoundException,
+} from "@nestjs/common";
+import {
+    AuthUserResult,
+    IAuthProvider,
+} from "../global/interfaces/oauth.interface";
+import { PrismaService } from "../prisma/prisma.service";
+import { GithubUser } from "../global/types/auth.types";
+import { generatePasswordHash } from "../global/auth/utils";
+import { OAuthConfig } from "@/config/Oauth/oauthConfig.interface";
+
+@Injectable()
+export class GithubAuthService implements IAuthProvider {
+    constructor(
+        private prisma: PrismaService,
+        @Inject("OAuth-Config") private oAuthConfig: OAuthConfig,
+    ) {}
+    async validateUser(user: GithubUser) {
```

Comment:
Just a tiny suggestion to make the intelliSense clearer, since Typescript was inferring unnecessarily a union type of objects with the same properties 
```suggestion
    async validateUser(user: GithubUser): Promise<AuthUserResult> {
```

Reply:
Is AuthUserResult something built in or something we have to define?
I think this might be same as the prisma User model type